gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```diff
@@ -53,113 +62,93 @@ public CaResources(ILogger logger) {
         this.logger = logger;
     }
 
-    public Optional<AliasContentCert> readPalantirRootCaFromSystemTruststore() {
-        return systemCertificates().flatMap(CaResources::selectPalantirCertificate);
+    public static Optional<AliasContentCert> readPalantirRootCaFromSystemTruststore() {
+        return selectPalantirCertificate(systemCertificates());
     }
 
-    public Optional<AliasContentCert> maybeGetCertificateFromSerialNumber(String serialNumber, String alias) {
-        return systemCertificates()
-                .map(certs -> selectCertificates(certs, Map.of(serialNumber, alias)))
-                .orElseGet(Stream::empty)
-                .findFirst();
+    public void importAllSystemCerts(Path jdkInstallationDirectory) {
+        importCertificates(jdkInstallationDirectory, parseCerts(systemCertificates()));
     }
 
-    public void importCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
-        Os os = CurrentOs.get();
-        switch (os) {
-            case MACOS:
-            case LINUX_GLIBC:
-            case LINUX_MUSL:
-                unixImportCertInJdk(aliasContentCert, jdkInstallationDirectory);
-                logger.log(String.format(
-                        "Successfully imported CA certificate %s into the JDK truststore",
-                        aliasContentCert.getAlias()));
-                break;
-            case WINDOWS:
-                logger.logError(
-                        String.format("Importing certificates for OS type '%s' is not yet supported", os.uiName()));
-                break;
-        }
-    }
-
-    private void unixImportCertInJdk(AliasContentCert aliasContentCert, Path jdkInstallationDirectory) {
+    private void importCertificates(Path jdkInstallationDirectory, List<Certificate> certificates) {
         try {
-            if (isCertificateInTruststore(jdkInstallationDirectory, aliasContentCert.getAlias())) {
-                logger.log("Certificate already exists in the truststore, skipping...");
-                return;
+            char[] passwd = "changeit".toCharArray();
+            Path jksPath = jdkInstallationDirectory.resolve("lib/security/cacerts");
+            KeyStore jks = loadKeystore(passwd, jksPath);
+            Set<BigInteger> existingSerialNumbers = getExistingCertificates(jks);
+            logger.log(String.format("Certificate %s imported successfully into the KeyStore.", existingSerialNumbers));
+            List<X509Certificate> newCertificates = certificates.stream()
+                    .map(X509Certificate.class::cast)
+                    .filter(certificate -> !existingSerialNumbers.contains(certificate.getSerialNumber()))
+                    .collect(Collectors.toList());
+            for (X509Certificate certificate : newCertificates) {
+                String alias = getAlias(certificate);
+                logger.log(String.format("Certificate %s imported successfully into the KeyStore.", alias));
```

Comment:
I'd say "...imported successfully into the JDK truststore from the system truststore" explicitly so people know where it came from.